From 3062c7e6b919c393846ff0e13558e7f648b352ca Mon Sep 17 00:00:00 2001
Message-Id: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: "Kumar, Kiran S" <kiran.s.kumar@intel.com>
Date: Wed, 28 Jan 2015 14:25:42 +0530
Subject: [PATCH 01/95] Revert "REVERTME [VPG]: drm/i915: disable maxfifo when
 in linear memory."

This reverts commit c407a4f7177e82f6a2ad9e4e28be57fffa56d5c0

As part of the atomicity, Maxfifo changes are taken care. Hence this patch is not needed.

Issue: GMINL-5637
Change-Id: I30f985f887fd06c4039f5f1eed8c157fb8a1b234
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h      |    1 -
 drivers/gpu/drm/i915/intel_display.c |   17 ++---------------
 drivers/gpu/drm/i915/intel_sprite.c  |   16 ++--------------
 3 files changed, 4 insertions(+), 30 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 58bf56c..d0bbc74 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1717,7 +1717,6 @@ struct drm_i915_private {
 
 	bool is_first_modeset;
 	bool maxfifo_enabled;
-	bool is_tiled;
 	bool gamma_enabled[I915_MAX_PIPES];
 	bool csc_enabled[I915_MAX_PIPES];
 	bool is_resuming;
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index e0ef5a6..1a2f8dd 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2780,23 +2780,10 @@ static void i9xx_update_primary_plane(struct drm_crtc *crtc,
 		rotate = true;
 
 	if (INTEL_INFO(dev)->gen >= 4) {
-		if (obj->tiling_mode != I915_TILING_NONE) {
+		if (obj->tiling_mode != I915_TILING_NONE)
 			dspcntr |= DISPPLANE_TILED;
-			dev_priv->is_tiled = true;
-		} else {
+		else
 			dspcntr &= ~DISPPLANE_TILED;
-			dev_priv->is_tiled = false;
-			/*
-			 * TODO:In linear mode disable maxfifo, hack to the
-			 * FADiag app flicker issue.
-			 */
-			if (dev_priv->maxfifo_enabled &&
-					!dev_priv->atomic_update) {
-				I915_WRITE(FW_BLC_SELF_VLV, ~FW_CSPWRDWNEN);
-				dev_priv->maxfifo_enabled = false;
-				intel_wait_for_vblank(dev, pipe);
-			}
-		}
 	}
 
 	if (IS_G4X(dev))
diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index 0300a2d..a9b71df 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -507,22 +507,10 @@ vlv_update_plane(struct drm_plane *dplane, struct drm_crtc *crtc,
 	 */
 	sprctl |= SP_GAMMA_ENABLE;
 
-	if (obj->tiling_mode != I915_TILING_NONE) {
+	if (obj->tiling_mode != I915_TILING_NONE)
 		sprctl |= SP_TILED;
-		dev_priv->is_tiled = true;
-	} else {
+	else
 		sprctl &= ~SP_TILED;
-		dev_priv->is_tiled = false;
-		/*
-		 * TODO:In linear mode disable maxfifo, hack to the
-		 * FADiag app flicker issue.
-		 */
-		if (dev_priv->maxfifo_enabled && !dev_priv->atomic_update) {
-			I915_WRITE(FW_BLC_SELF_VLV, ~FW_CSPWRDWNEN);
-			dev_priv->maxfifo_enabled = false;
-			intel_wait_for_vblank(dev, pipe);
-		}
-	}
 
 	sprctl |= SP_ENABLE;
 
-- 
1.7.9.5

